packages/core/src/common/config-manager.ts

```diff
@@ -0,0 +1,182 @@
+import ts from 'typescript';
+import { fileURLToPath } from 'node:url';
+import { dirname, extname, resolve } from 'node:path';
+import type { Options as PrettierOptions } from 'prettier';
+import { createRequire } from 'node:module';
+
+export type TsUserConfigLang = 'typescript' | 'javascript';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+const require = createRequire(import.meta.url);
+
+export interface TSUserConfig {
+  preferences?: TsUserPreferencesConfig;
+  suggest?: TSSuggestConfig;
+  format?: TsFormatConfig;
+}
+
+export interface TsUserPreferencesConfig {
+  importModuleSpecifier: ts.UserPreferences['importModuleSpecifierPreference'];
+  importModuleSpecifierEnding: ts.UserPreferences['importModuleSpecifierEnding'];
+  quoteStyle: ts.UserPreferences['quotePreference'];
+  /**
+   * only in typescript config
+   */
+  includePackageJsonAutoImports?: ts.UserPreferences['includePackageJsonAutoImports'];
+}
+
+export type TsFormatConfig = Omit<
+  ts.FormatCodeSettings,
+  'indentMultiLineObjectLiteralBeginningOnBlankLine' | keyof ts.EditorSettings
+>;
+
+export interface TSSuggestConfig {
+  autoImports: ts.UserPreferences['includeCompletionsForModuleExports'];
+  includeAutomaticOptionalChainCompletions: boolean | undefined;
+  includeCompletionsForImportStatements: boolean | undefined;
+}
+
+export class ConfigManager {
+  private prettierConfig: PrettierOptions = {};
+  updateTsJsUserPreferences(): void {}
+  updateTsJsFormatConfig(config: Record<TsUserConfigLang, TSUserConfig>): void {
+    (['typescript', 'javascript'] as const).forEach((lang) => {
+      if (config[lang]) {
+        this.updateTsFormatConfig(lang, config[lang]);
+      }
+    });
+  }
+
+  private updateTsFormatConfig(lang: TsUserConfigLang, config: TSUserConfig): void {
+    this.tsFormatCodeOptions[lang] = {
+      ...this.tsFormatCodeOptions[lang],
+      ...(config.format ?? {}),
+    };
+  }
+
+  private tsFormatCodeOptions: Record<TsUserConfigLang, ts.FormatCodeSettings> = {
+    javascript: this.getDefaultFormatCodeOptions(),
+    typescript: this.getDefaultFormatCodeOptions(),
+  };
+
+  getFormatCodeSettingsForFile(filePath: string): ts.FormatCodeSettings {
+    const extension = extname(filePath);
+    const configLang =
+      extension === 'ts' || extension === 'gts' || extension === 'tsx'
+        ? 'typescript'
+        : 'javascript';
+
+    const prettierConfig = this.getMergedPrettierConfig(
+      importPrettier(filePath).resolveConfig.sync(filePath, {
+        editorconfig: true,
+      })
+    );
```

Comment:
I think there is a question of if we should do this or not. This is just here to support non-vscode users since I believe  [this](https://github.com/typed-ember/glint/pull/499/files#diff-b07d9e1ce11d8e806c911933f186c2955f9d8e94aaea85cf349cc98a8a088255) will just pull it from the workspace if the prettier plugin is installed.